File: RQ_RLMI/Range.py
class Range:
    def __init__(self, rangeConfig):
        self.LOW = rangeConfig["LOW"]
        self.HIGH = rangeConfig["HIGH"]
        self.ID = rangeConfig["ID"]

    def __str__(self):
        return "Range %d: [%d, %d)" % (self.ID, self.LOW, self.HIGH+1)

    def matchPos(self, pos) -> bool:
        if self.LOW <= pos <= self.HIGH:
            return True
        else:
            return False

File: RQ_RLMI_CPP/src/Range.cpp
#include "Range.h"

Range::Range(unsigned long long int _Low, unsigned long long int _High, unsigned long long int _id) {
    this->Low = _Low;
    this->High = _High;
    this->id = _id;
}

void Range::printSelf() {
    printf("Range %d: [%d, %d)", id, Low, High+1);
}

//bool Range::match(unsigned int pos) {
//    if(pos >= this->Low && pos <= this->High) {
//        return true;
//    } else {
//        return false;
//    }
//}